### ISSUE TITLE
[Bug]: Netdata uninstall doesn't work on Arch Linux

### ISSUE THREAD

**Issue by IanTrudel** [netdata/netdata#16119]
### Bug description

```
 --- Using /tmp/netdata-kickstart-bfrzH9EIHr as a temporary directory. ---
 --- Checking for existing installations of Netdata... ---
 --- Downloading netdata-uninstaller ... ---
[/tmp/netdata-kickstart-bfrzH9EIHr]$ /usr/bin/curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-bfrzH9EIHr/netdata-uninstaller.sh https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/netdata-uninstaller.sh
 OK

[/tmp/netdata-kickstart-bfrzH9EIHr]$ sudo /tmp/netdata-kickstart-bfrzH9EIHr/netdata-uninstaller.sh --yes
[sudo] Password for username :
/tmp/netdata-kickstart-bfrzH9EIHr/netdata-uninstaller.sh: ligne 711: /etc/netdata/.environment: Aucun fichier ou dossier de ce type

mar 03 oct 2023 19:09:17 EDT : FATAL: netdata-uninstaller.sh: FAILED TO UNINSTALL NETDATA:  Failed to completely remove Netdata from this system.
 FAILED

Finished uninstalling the Netdata Agent.The following non-fatal warnings or errors were encountered:

  - Command "sudo /tmp/netdata-kickstart-bfrzH9EIHr/netdata-uninstaller.sh --yes" failed with exit code 1.

Root privileges required to run rm -rf /tmp/netdata-kickstart-bfrzH9EIHr
[/home/username]$ sudo rm -rf /tmp/netdata-kickstart-bfrzH9EIHr
 OK
```

### Expected behavior

The uninstaller should not fail and Netdata should be removed entirely.

### Steps to reproduce

1. `curl https://my-netdata.io/kickstart.sh > /tmp/netdata-kickstart.sh && sh /tmp/netdata-kickstart.sh --uninstall`
2. Enter password for sudo.

### Installation method

kickstart.sh

### System info

```shell
Arch Linux
```

### Netdata build info

```shell
Packaging:
    Netdata Version ____________________________________________ : v1.42.4
    Installation Type __________________________________________ : kickstart-static
    Package Architecture _______________________________________ : x86_64
    Package Distro _____________________________________________ : unknown
    Configure Options __________________________________________ :  '--prefix=/opt/netdata/usr' '--sysconfdir=/opt/netdata/etc' '--localstatedir=/opt/netdata/var' '--libexecdir=/opt/netdata/usr/libexec' '--libdir=/opt/netdata/usr/lib' '--with-zlib' '--with-math' '--with-user=netdata' '--enable-cloud' '--without-bundled-protobuf' '--disable-dependency-tracking' '--enable-lto' 'CFLAGS=-ffunction-sections -fdata-sections -static -O2 -funroll-loops -I/openssl-static/include -I/libnetfilter-acct-static/include/libnetfilter_acct -I/usr/include/libmnl -pipe' 'LDFLAGS=-Wl,--gc-sections -Wl,-z,stack-size=8388608 -static -L/openssl-static/lib64 -L/libnetfilter-acct-static/lib -lnetfilter_acct -L/usr/lib -lmnl' 'PKG_CONFIG=pkg-config --static' 'PKG_CONFIG_PATH=/openssl-static/lib64/pkgconfig:/libnetfilter-acct-static/lib/pkgconfig:/usr/lib/pkgconfig'
Default Directories:
    User Configurations ________________________________________ : /opt/netdata/etc/netdata
    Stock Configurations _______________________________________ : /opt/netdata/usr/lib/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /opt/netdata/var/cache/netdata
    Permanent Databases ________________________________________ : /opt/netdata/var/lib/netdata
    Plugins ____________________________________________________ : /opt/netdata/usr/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /opt/netdata/usr/share/netdata/web
    Log Files __________________________________________________ : /opt/netdata/var/log/netdata
    Lock Files _________________________________________________ : /opt/netdata/var/lib/netdata/lock
    Home _______________________________________________________ : /opt/netdata/var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : Linux
    Kernel Version _____________________________________________ : 5.18.6
    Operating System ___________________________________________ : Arch Linux
    Operating System ID ________________________________________ : arch
    Operating System ID Like ___________________________________ : unknown
    Operating System Version ___________________________________ : rolling
    Operating System Version ID ________________________________ : none
    Detection __________________________________________________ : Mixed
Hardware:
    CPU Cores __________________________________________________ : 
    CPU Frequency ______________________________________________ : 3400000000
    CPU Architecture ___________________________________________ : 33622183936
    RAM Bytes __________________________________________________ : 500118700032
    Disk Capacity ______________________________________________ : x86_64
    Virtualization Technology __________________________________ : none
    Virtualization Detection ___________________________________ : systemd-detect-virt
Container:
    Container __________________________________________________ : none
    Container Detection ________________________________________ : systemd-detect-virt
    Container Orchestrator _____________________________________ : none
    Container Operating System _________________________________ : none
    Container Operating System ID ______________________________ : none
    Container Operating System ID Like _________________________ : none
    Container Operating System Version _________________________ : none
    Container Operating System Version ID ______________________ : none
    Container Operating System Detection _______________________ : none
Features:
    Built For __________________________________________________ : Linux
    Netdata Cloud ______________________________________________ : YES
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (lz4)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
Database Engines:
    dbengine ___________________________________________________ : YES
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    map ________________________________________________________ : YES
    save _______________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : YES
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : YES
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    Judy (high-performance dynamic arrays and hashtables) ______ : YES (bundled)
    dlib (robust machine learning toolkit) _____________________ : YES (bundled)
    protobuf (platform-neutral data serialization protocol) ____ : YES (system)
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : NO
    libcrypto (cryptographic functions) ________________________ : YES
    libm (mathematical functions) ______________________________ : YES
    jemalloc ___________________________________________________ : NO
    TCMalloc ___________________________________________________ : NO
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : YES
    cgroup-network (associate interfaces to CGROUPS) ___________ : YES
    proc (monitor Linux systems) _______________________________ : YES
    tc (monitor Linux network QoS) _____________________________ : YES
    diskspace (monitor Linux mount points) _____________________ : YES
    freebsd (monitor FreeBSD systems) __________________________ : NO
    macos (monitor MacOS systems) ______________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : YES
    cups (monitor printers and print jobs) _____________________ : NO
    ebpf (monitor system calls) ________________________________ : YES
    freeipmi (monitor enterprise server H/W) ___________________ : NO
    nfacct (gather netfilter accounting) _______________________ : YES
    perf (collect kernel performance events) ___________________ : YES
    slabinfo (monitor kernel object caching) ___________________ : YES
    Xen ________________________________________________________ : NO
    Xen VBD Error Tracking _____________________________________ : NO
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : NO
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : YES
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : NO
```

**Comment by ktsaou** [comment]
@Ferroin @tkatsoulas the uninstaller looks for `/etc/netdata/.environment`, but this is static install, so it should look at `/opt/netdata/etc/netdata/.environment`.

**Comment by IanTrudel** [comment]
> @Ferroin @tkatsoulas the uninstaller looks for `/etc/netdata/.environment`, but this is static install, so it should look at `/opt/netdata/etc/netdata/.environment`.

I can confirm that the file is indeed located at `/opt/netdata/etc/netdata/.environment` on my machine.

**Comment by ktsaou** [comment]
@IanTrudel since I know you were interested for journal logs, unfortunately this kind of installation does not support them. Check here: https://github.com/netdata/netdata/tree/master/collectors/systemd-journal.plugin#limitations

Now, on Arch there is a native package there, but it is not the nightly version of Netdata. So, this plugin is not there yet.

So, to tell you the truth, unless you are prepared to install netdata from source, I am not sure how you can test this on your arch system. It is still one command (I use manjaro myself) to install it from source, but a lot of development libraries and tools will be installed on your system.

If you have a debian, ubuntu, etc test VM, for which we have native packages this would be very easy.

**Comment by IanTrudel** [comment]
@ktsaou Netdata works really well so far. I followed the installation instructions available in the [documentation](https://learn.netdata.cloud/docs/installing/one-line-installer-for-all-linux-systems). I just had the idea to try out the uninstaller and it failed. Thought you guys would like to know. I can also test again if a new uninstaller comes out and report on it.

**Comment by IanTrudel** [comment]
Here's a temporary solution, if anyone needs it:

```zsh
sudo ln -s /opt/netdata/etc/netdata /etc/netdata 
curl https://my-netdata.io/kickstart.sh > /tmp/netdata-kickstart.sh && sh /tmp/netdata-kickstart.sh --uninstall 
```

**Comment by ilyam8** [comment]
>  --- Downloading netdata-uninstaller ... ---

The OP issue is the following: 
- kickstart couldn't find `netdata-uninstaller.sh` on the host system and
- downloaded it from the netdata repo.
- the one in the repo didn't have the correct `ENVIRONMENT_FILE` path (expected).

So `INSTALL_PREFIX` wasn't set (detected) during [detect_existing_install()](https://github.com/netdata/netdata/blob/499e1923ebacd7aaf1214ecc21f7a1e6c930829e/packaging/installer/kickstart.sh#L846C1-L846C24).

@IanTrudel 

- Have you previously installed Netdata using pacman?
- Can you show `pacman -Qi netdata` output and exit code?
- Do you have `/etc/netdata/` dir with files?

**Comment by IanTrudel** [comment]
> @IanTrudel
> 
> * Have you previously installed Netdata using pacman?
> * Can you show `pacman -Qi netdata` output and exit code?
> * Do you have `/etc/netdata/` dir with files?

@ilyam8 

* No, I installed it according to what was in the documentation for all Linux distro: `curl https://my-netdata.io/kickstart.sh > /tmp/netdata-kickstart.sh && sh /tmp/netdata-kickstart.sh`.
* No, see above.
* `/opt/netdata/etc/netdata/`

**Comment by ilyam8** [comment]
@IanTrudel found the issue, fixed in #15146
